### ISSUE TITLE
Postgres Search Plug-in keeping backend from starting

### ISSUE THREAD

**Issue by zjpersc** [backstage/backstage#6861]
While implementing release-08-12 in our sandbox environment, we are unable to get the Backstage backend to start. The log provides the following output:

```
2021-08-17T20:34:48.761Z auth info Configuring provider, microsoft type=plugin
migration file "20210311110000_init.js" failed
migration failed with error: create table "documents" ("type" text not null, "document" jsonb not null, "body" tsvector NOT NULL GENERATED ALWAYS AS (setweight(to_tsvector('english', document->>'title'), 'A') || setweight(to_tsvector('english', document->>'text'), 'B') || setweight(to_tsvector('english', document - 'location' - 'title' - 'text'), 'C')) STORED) - syntax error at or near "("
Backend failed to start up error: create table "documents" ("type" text not null, "document" jsonb not null, "body" tsvector NOT NULL GENERATED ALWAYS AS (setweight(to_tsvector('english', document->>'title'), 'A') || setweight(to_tsvector('english', document->>'text'), 'B') || setweight(to_tsvector('english', document - 'location' - 'title' - 'text'), 'C')) STORED) - syntax error at or near "("
    at Parser.parseErrorMessage (C:\dev\sf_backstage_local\node_modules\pg-protocol\dist\parser.js:287:98)
    at Parser.handlePacket (C:\dev\sf_backstage_local\node_modules\pg-protocol\dist\parser.js:126:29)
    at Parser.parse (C:\dev\sf_backstage_local\node_modules\pg-protocol\dist\parser.js:39:38)
    at Socket.<anonymous> (C:\dev\sf_backstage_local\node_modules\pg-protocol\dist\index.js:11:42)
    at Socket.emit (events.js:315:20)
    at Socket.EventEmitter.emit (domain.js:467:12)
    at addChunk (internal/streams/readable.js:309:12)
    at readableAddChunk (internal/streams/readable.js:284:9)
    at Socket.Readable.push (internal/streams/readable.js:223:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:188:23) {
  length: 91,
  severity: 'ERROR',
  code: '42601',
  detail: undefined,
  hint: undefined,
  position: '121',
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'scan.l',
  line: '1134',
  routine: 'scanner_yyerror'
}
```

## Expected Behavior

Backstage backend should start

## Current Behavior

The error message above is displayed in the log and aborts start-up. 

## Possible Solution

Not a solution, but a workaround is to comment out the Postgres Search plugin in packages/backend/src/plugins/search.ts (line 24;50-52)

## Steps to Reproduce

1. Download and run release-08-12 locally
2. Start the Backstage backend

## Your Environment
- NodeJS Version: v14.16.0 
- Operating System and Version: Windows 10 and node-14:buster image, separate systems, same result
- Browser Information: Chrome 92.0.4515.131
- Postgres 11.12

**Comment by freben** [comment]
Oh, @Fox32  may be interested here.

https://github.com/backstage/backstage/blob/master/plugins/search-backend-module-pg/src/database/DatabaseDocumentStore.ts#L62

https://www.postgresql.org/docs/12/ddl-generated-columns.html

Seems to me that it's the version check that needs to be updated to 12, not 11.

@zjperjd you have the ability to run against version 12 or newer?

**Comment by Fox32** [comment]
Oh that is unfortunate. Looks like I messed up the version number there. It's not just the check, we also have to update the docs. Maybe something we should mention at the meetup tomorrow.

**Comment by Fox32** [comment]
I created a PR to fix the version requirement #6863.

**Comment by zjpersc** [comment]
I verified that running against Postgres v12 allows the code to be uncommented and the backend to start successfully.
